DRILL-6640: Drill takes long time in planning when there are large number of files in views/tables DFS parent directory

kr-arjun opened the pull request:
Includes
1.  Modified DotDrillUtil.getDotDrills implementation to avoid using FS globStatus call with GLOB for a given base file name.
2. Added unit test cases for the new method.
3. Code refactoring to include additional comments.

HanumathRao commented:
@kr-arjun There seems to be some test failures. Can you verify if these test failures are because of your latest code.

kr-arjun commented:
@HanumathRao I have fixed the test failures. Could you please review now.

priteshm commented:
@HanumathRao can you complete the review today?

HanumathRao commented on exec/java-exec/src/main/java/org/apache/drill/exec/dotdrill/DotDrillUtil.java:
```diff
@@ -48,16 +59,70 @@
+        statusList.addAll(Arrays.asList(status));
+      }
+    } else {
+      DotDrillType[] typeArray;
```
Instead of typeArray you may want  to use "types".

HanumathRao commented on exec/java-exec/src/main/java/org/apache/drill/exec/dotdrill/DotDrillUtil.java:
```diff
@@ -48,16 +59,70 @@
+    List<FileStatus> statusList = new ArrayList<FileStatus>();
+
+    if (name.endsWith(".drill")) {
+      FileStatus[] status = fs.globStatus(new Path(root, name));
```
Why do you want to change the existing logic for this case?

HanumathRao replied:
In this case does it mean that types should not matter? If so is it good to have some assert.

HanumathRao commented on exec/java-exec/src/main/java/org/apache/drill/exec/dotdrill/DotDrillUtil.java:
```diff
@@ -48,16 +59,70 @@
+   * @throws IOException  if any I/O error occurs when fetching file status
+   */
+  private static List<FileStatus> getDrillFileStatus(DrillFileSystem fs, Path root, String name, DotDrillType... types) throws IOException{
+    List<FileStatus> statusList = new ArrayList<FileStatus>();
```
filesStatus does it make sense?

HanumathRao commented on exec/java-exec/src/test/java/org/apache/drill/exec/dotdrill/TestDotDrillUtil.java:
```diff
@@ -0,0 +1,100 @@
+
+public class TestDotDrillUtil {
+
+  private static final String TEST_ROOT_DIR = System.getProperty("java.io.tmpdir") + "/dot_drill_util";
```
I think you may want to look at some existing tests to inherit baseclass like BaseTestQuery. This will make sure that the test directories are cleaned up properly.

@ilooner  Can you please take a look at the test file.

kr-arjun replied:
@ilooner Could you please review the Test file. This test scenario is more related to FileSystem related tests. Please let me know if there are any existing base class that can be used here.

kr-arjun commented:
@HanumathRao  I have modified variable names as per review comments. I didn't include assert check for types length while checking name ending with '.drill' as it is causing test failures. This assertion check fails for the scenario where complete view name is used in querying as in below test.

https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/exec/sql/TestViewSupport.java#L748

 For '*.drill' files, the type check is being done later in the getDrillFiles method. I have modified the method comments to indicate this behavior that types checking is done only for case name does not end with .drill. Please let me know if this is okay.

HanumathRao commented:
@kr-arjun Thanks for the changes. Code changes look good to me.

kr-arjun commented:
Thanks, @HanumathRao . @ilooner  Could you please review these changes?

priteshm commented:
@ilooner can you please review this as well?

kr-arjun commented:
@ilooner I found a possible issue with current logic. It may fail if the path has a wildcard ( case where not ending with .drill. In this case list status may fail). I am pushing the changes for handling this scenario. Thanks!

kr-arjun commented:
@ilooner  I have pushed the latest changes. Could you please review. Thanks!

ilooner commented:
Sorry for the delayed review @kr-arjun. Please use BaseDirTestWatcher for creating temp directories in tests. There is information in the javadoc for the class, an example in ExampleTest, and you can take a look at the other tests that use it for more examples.

arjun-rajan commented:
@ilooner  I have made the changes as per the review comment. Could you please review. Thanks!

kr-arjun commented:
@ilooner  I have made the changes as per the review comment. Could you please review. Thanks!

arjun-rajan commented:
@ilooner Could you please review the changes.

arina-ielchiieva commented:
@kr-arjun please squash commits into one.

kr-arjun commented:
@arina-ielchiieva I have squashed the commits into one. Thanks!

sohami commented:
@kr-arjun - Please make the PR and JIRA title same.

kr-arjun commented:
@sohami I have corrected the PR title. Thanks!
